src/libstore/build/worker.cc

```diff
@@ -187,7 +194,10 @@ void Worker::childStarted(GoalPtr goal, const std::set<int> & fds,
     child.inBuildSlot = inBuildSlot;
     child.respectTimeouts = respectTimeouts;
     children.emplace_back(child);
-    if (inBuildSlot) nrLocalBuilds++;
+    if (inBuildSlot) {
+        if (dynamic_cast<PathSubstitutionGoal *>(child.goal2)) nrSubstitutions++;
```

Comment:
This should also treat `DrvOutputSubstitutionGoal` as a substitution, since it's network bound.

Rather than using `dynamic_cast`, it's probably better to have a virtual method `isSubstitution()` that returns true for `PathSubstitutionGoal` and `DrvOutputSubstitutionGoal`, and false for the other goals.

Reply:
Perhaps even have a

```c++
/**
 * What quota does this job count towards?
 */
enum struct JobCategory {
  Build,
  Substitution,
};
```
and then a method that return the job category.